test_restore_db_replica/test.py::test_restore_db_replica_with_diffrent_table_metadata is flaky

https://play.clickhouse.com/play?user=play#CldJVEgKICAgICd0ZXN0X3Jlc3RvcmVfZGJfcmVwbGljYV93aXRoX2RpZmZyZW50X3RhYmxlX21ldGFkYXRhJyBBUyBuYW1lX3N1YnN0ciwKICAgIDMwIEFTIGludGVydmFsX2RheXMKU0VMRUNUCiAgICB0ZXN0X25hbWUsCiAgICBjaGVja19zdGFydF90aW1lIEFTIGQsCiAgICBwdWxsX3JlcXVlc3RfbnVtYmVyIEFTIHBycywKICAgIHJlcG9ydF91cmwsCkZST00gY2hlY2tzCldIRVJFICgobm93KCkgLSB0b0ludGVydmFsRGF5KGludGVydmFsX2RheXMpKSA8PSBjaGVja19zdGFydF90aW1lKSBBTkQgKHBvc2l0aW9uKHRlc3RfbmFtZSwgbmFtZV9zdWJzdHIpID4gMCkgQU5EICh0ZXN0X3N0YXR1cyBJTiAoJ0ZBSUwnLCAnRVJST1InLCAnRkxBS1knKSkKT1JERVIgQlkgZCBERVNDCg==

Example failure:
```
        if restore_firstly_node_where_created:
>           assert node_1.query(
                f"SELECT count(*) FROM {exclusive_database_name}.{test_table_2}"
            ) == TSV([count_test_table_2])
E           AssertionError: assert '7\n' == 10
E            +  where '7\n' = <bound method ClickHouseInstance.query of <helpers.cluster.ClickHouseInstance object at 0x7f74aa23c0a0>>('SELECT count(*) FROM repl_db_test_restore_db_replica_with_diffrent_table_metadata_restore_node2_node1__oqzqml.test_table_2')
E            +    where <bound method ClickHouseInstance.query of <helpers.cluster.ClickHouseInstance object at 0x7f74aa23c0a0>> = <helpers.cluster.ClickHouseInstance object at 0x7f74aa23c0a0>.query
E            +  and   10 = TSV([10])

test_restore_db_replica/test.py:512: AssertionError
```

--- a/tests/integration/test_restore_db_replica/test.py
+++ b/tests/integration/test_restore_db_replica/test.py
@@ -535,12 +535,12 @@ def test_restore_db_replica_with_diffrent_table_metadata(
     )
 
     if restore_firstly_node_where_created:
-        assert node_1.query(
-            f"SELECT count(*) FROM {exclusive_database_name}.{test_table_2}"
-        ) == TSV([count_test_table_2]) 
-        assert node_2.query(
-            f"SELECT count(*) FROM {exclusive_database_name}.{test_table_2}"
-        ) == TSV([count_test_table_2]) 
+        check_contains_table(
+            node_1, f"{exclusive_database_name}.{test_table_2}", count_test_table_2
+        )
+        check_contains_table(
+            node_2, f"{exclusive_database_name}.{test_table_2}", count_test_table_2
+        )
     else:
         assert node_2.query(
             f"SELECT count(*) FROM system.databases WHERE name='{exclusive_database_name}_broken_tables'"

